### PR TITLE
flow: add a test case using free boundary conditions

### PR DESCRIPTION
**Pull request by andlaus**
this is based on case 1 of SPE-1.

### CODE REVIEW

**Comment by andlaus** [comment]
jenkins build this opm-tests=92 please

**Comment by akva2** [comment]
needs to be added in tests/update_reference_data.sh.

File: tests/update_reference_data.sh [hunk 1]
```diff
@@ -37,6 +37,7 @@ tests[spe12p]="flow spe1 SPE1CASE2_2P spe1_2p"
 tests[spe1oilgas]="flow spe1 SPE1CASE2_OILGAS spe1_oilgas"
 tests[spe1nowells]="flow spe1 SPE1CASE2_NOWELLS spe1_nowells"
 tests[spe1thermal]="flow spe1 SPE1CASE2_THERMAL spe1_thermal"
+tests[spe1thermal]="flow spe1 SPE1CASE1_FREEBC spe1_freebc"
```
Comment (akva2) [inline tests/update_reference_data.sh]: unique key please.

**Comment by andlaus** [comment]
> needs to be added in tests/update_reference_data.sh.

thanks. this is not obvious at all. should be fixed, can you confirm?

File: tests/update_reference_data.sh (hunk 1 above)
Comment (andlaus) [reply tests/update_reference_data.sh]: what do you mean? (I copied the line above and replaced "thermal" by "freebc".)

Comment (andlaus) [reply tests/update_reference_data.sh]: ah, I see. forgot the LHS...

Comment (andlaus) [reply tests/update_reference_data.sh]: should be fixed. it is possibly worth to think about modifying the build system to do this automatically. ( `opm_add_test()` should already get all required information.)

**Comment by andlaus** [comment]
jenkins build this opm-tests=92 please

File: tests/update_reference_data.sh (hunk 1 above)
Comment (akva2) [reply tests/update_reference_data.sh]: i don't think so. this is not related to the build system, nor is it a task for the build system. this is related to jenkins automation only.

Comment (andlaus) [reply tests/update_reference_data.sh]: yes, but the build system could generate the update script. Anyway, this is (mostly) not my cup of tea...

Comment (alfbr) [reply tests/update_reference_data.sh]: Just want to add that not all results are automatically updated. The cell volume calculations need to be manually updated, and that is deliberate.

**Comment by totto82** [comment]
I indent to add a test for mass flow bc to this test and rename it to spe1_bc. Can I close this and make a new one or should I merge this and make a new PR.

**Comment by andlaus** [comment]
from my POV: great, let's close this one.
